### ISSUE TITLE
Using the suffix "Value" conflicts with generated enum getters/setters

### ISSUE THREAD

**Issue by zackthehuman** [protocolbuffers/protobuf#15411]
**What version of protobuf and what language are you using?**
Version: 3.19.4

```
> protoc --version
libprotoc 3.19.4
```

Language: Java

**What operating system (Linux, Windows, ...) and version?**
macOS 14.2.1 (23C71)

**What runtime / compiler are you using (e.g., python version or gcc version)**
11.0.13-msft

**What did you do?**
When a message has a field that is an enum, two getters and setters are generated -- one for the enum type and another for the enum's value which accepts an `int` and has the suffix "Value" (e.g. `MyEnum getFoo()` and `int getFooValue()`). This can lead to generating invalid Java code if another field in the message has the same name as the enum field + "Value".

Define an enum and message:
```proto
syntax = "proto3";

package example;

option java_multiple_files = true;
option java_package = "com.example";
option java_outer_classname = "DoubleGettingExamples";

enum Property {
  Property_FOO = 0;
  Property_BAR = 1;
  Property_BAZ = 2;
}

message DoubleGettingExample {
  Property example = 1;
  string exampleValue = 2; // string example_value = 2; also has this problem because the codegen is the same.
}
```

Generate Java code:
```
protoc -I=/Users/me/example --java_out=/Users/me/example /Users/me/example/DoubleGetterExample.proto 
```

Resulting Java class has two different `getExampleValue` methods that differ in return type, which doesn't compile:
```java
public static final int EXAMPLE_FIELD_NUMBER = 1;
private int example_ = 0;
/**
 * <code>.example.Property example = 1;</code>
 * @return The enum numeric value on the wire for example.
 */
@java.lang.Override public int getExampleValue() {
  return example_;
}
/**
 * <code>.example.Property example = 1;</code>
 * @return The example.
 */
@java.lang.Override public com.linkedin.example.Property getExample() {
  com.linkedin.example.Property result = com.linkedin.example.Property.forNumber(example_);
  return result == null ? com.linkedin.example.Property.UNRECOGNIZED : result;
}

public static final int EXAMPLEVALUE_FIELD_NUMBER = 2;
@SuppressWarnings("serial")
private volatile java.lang.Object exampleValue_ = "";
/**
 * <code>string exampleValue = 2;</code>
 * @return The exampleValue.
 */
@java.lang.Override
public java.lang.String getExampleValue() {
  java.lang.Object ref = exampleValue_;
  if (ref instanceof java.lang.String) {
    return (java.lang.String) ref;
  } else {
    com.google.protobuf.ByteString bs = 
        (com.google.protobuf.ByteString) ref;
    java.lang.String s = bs.toStringUtf8();
    exampleValue_ = s;
    return s;
  }
}
```

1. Define an enum and a message.
2. Use the enum in the message.
3. Add another field to the message that has the same name as the enum field + `Value` at the end. The pattern of enum field + `_value` also has this problem.
4. Compile the `proto` to Java.
5. Observe that multiple getters with the same name are generated.

**What did you expect to see**
I was hoping that a valid `proto` will result in the generation of valid Java source code. In some field name collision cases, `protoc` can handle it by renaming the generated getters/setters, but in this case it seems like it doesn't.

**What did you see instead?**
The resulting Java code is not valid since it generates a class with duplicate methods that differ in their return type, which is not allowed by the Java compiler.

**Anything else we should know about your project / environment**
We are currently working on a project to automatically convert data models from another format to `proto`, but sometimes the original fields have this kind of pattern, which breaks the resulting code. As a result, we have to work around this by mangling the perfectly good names of fields during conversion to `proto`.

**Comment by acozzette** [comment]
Sorry for the inconvenience, but unfortunately we're not able to prioritize fixing this right now. Probably the best workaround is to tweak the names in the proto file to keep them from conflicting in the generated code.

**Comment by zackthehuman** [comment]
@acozzette Thanks for the response. I totally understand. For now, we're working around this in two ways: 1) when translating schemas to `proto`, we "sanitize" names of fields that cause this problem by mangling their names -- ugly, but effective. 2) We've added in-house linting validation that prohibits naming fields in a way that conflict with enum fields when such enum fields exist in the schema.

In terms of a fix at the protobuf level, any idea _how_ this would be addressed? For example, would we append the field number to the getter/setter, similar to how other conflicts are handled? If we have some idea of _how_ we could handle this, then it may be possible to have a fix contributed rather than left on your plate.

**Comment by acozzette** [comment]
I think the fix would have to detect when this conflict between `x` and `x_value` happens, and then adjust the generated names to prevent them from conflicting. (If `x_value` appears by itself then we wouldn't want to mess with it because that might break existing code that works.) The question then is just what the adjustment should look like. I wasn't aware that we append the field number to address similar conflicts, but if so then that sounds like a reasonable precedent to follow.

**Comment by sebek64** [comment]
The same kind of conflict can happen for oneof and its "Case" suffixed getters, and all other artificial class properties.

**Comment by khaledh** [comment]
For anyone who stumbles upon this issue, it has been fixed since v4.30.0 (https://github.com/protocolbuffers/protobuf/pull/19117). Hopefully a maintainer can mark this issue as resolved instead of won't fix.
